_posts/2018/2018-03-18-check-your-database-indexes-for-activerecord-scopes.md

```diff
@@ -0,0 +1,59 @@
+---
+title: 'Check Your Database Indexes For ActiveRecord Scopes'
+description: 'Database indexes are always more important than you think'
+layout: article
+category: ruby
+image:
+  base: '2018/check-your-database-indexes-for-activerecord-scopes'
+  alt: 'Construction Site'
+  source: 'https://unsplash.com/photos/8Gg2Ne_uTcM'
+  credit: '贝莉儿 NG'
+---
+
+It is alleged that Ruby is slow. However, whether you're using Ruby, Elixir or Go, the 'benchmarked' speed of a language is irrelevant if your web application eventually hits an unoptimised database.
+
+
+## Instead of…
+
+...having your database do more work than it needs to.
+
+
+## Use…
+
+...database indexes when you query on foreign keys and for queries over multiple fields.
```

Comment:
I think what's missing is a simple explanation of what indexes are. That should go in this 'Use' section.

Then your example will be clearer. For example, line 28 below: you can now more search the 'model_with_foreign_key' by user_id more efficiently, right?

Reply:
This is brilliant feedback. You're so right.